internal/pluginframework/tfschema/customizable_schema.go

```diff
@@ -0,0 +1,153 @@
+package tfschema
+
+import (
+	"fmt"
+
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+// CustomizableSchema is a wrapper struct on top of AttributeBuilder that can be used to navigate through nested schema add customizations.
+type CustomizableSchema struct {
+	attr AttributeBuilder
+}
+
+// ConstructCustomizableSchema constructs a CustomizableSchema given a map from string to AttributeBuilder.
+func ConstructCustomizableSchema(attributes map[string]AttributeBuilder) *CustomizableSchema {
+	attr := AttributeBuilder(SingleNestedAttributeBuilder{Attributes: attributes})
+	return &CustomizableSchema{attr: attr}
+}
+
+// ToAttributeMap converts CustomizableSchema into a map from string to Attribute.
+func (s *CustomizableSchema) ToAttributeMap() map[string]AttributeBuilder {
+	return attributeToMap(&s.attr)
+}
+
+// attributeToMap converts AttributeBuilder into a map from string to AttributeBuilder.
+func attributeToMap(attr *AttributeBuilder) map[string]AttributeBuilder {
+	var m map[string]AttributeBuilder
+	switch attr := (*attr).(type) {
+	case SingleNestedAttributeBuilder:
+		m = attr.Attributes
+	case ListNestedAttributeBuilder:
+		m = attr.NestedObject.Attributes
+	case MapNestedAttributeBuilder:
+		m = attr.NestedObject.Attributes
+	default:
+		panic(fmt.Errorf("cannot convert to map, attribute is not nested"))
+	}
+
+	return m
+}
+
+func (s *CustomizableSchema) AddValidator(v any, path ...string) *CustomizableSchema {
```

Comment:
I tried to find a common interface for all of the validators but it seems like it's hard because each one of these has a separate method like `ValidateFloat64` or `ValidateBool` [link](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.11.0/schema/validator#Float64)

Reply:
@mgyucht maybe I missed something here, let me know if you have ideas, thanks!

Reply:
Nope, this is the right way to do this. This is the reason we couldn't include the AddValidator method in the AttributeBuilder interface.